src/hooks/useRampQuotes.ts

```diff
@@ -55,7 +56,10 @@ export const useRampQuotes = ({
           Result<RampQuote[], QuoteError>
         > => {
           try {
-            const quotes = await plugin.fetchQuotes(rampQuoteRequest)
+            const quotes = await runWithTimeout(
+              plugin.fetchQuotes(rampQuoteRequest),
```

Comment:
The core has a fancy-pants way of dealing with this, but for this PR we can just do a simple timeout.